input/musig2_session_manager.go

```diff
@@ -274,6 +280,12 @@ func (m *MusigSessionManager) MuSig2RegisterNonces(sessionID MuSig2SessionID,
 			sessionID[:])
 	}
 
+	// Check if combined nonce was already registered.
+	if session.hasCombinedNonce {
```

Comment:
nit: we don't necessarily need this check as `session.HaveAllNonces` will also be true at this point. If you agree we could just remove `hasCombinedNonce` altogether.